src/staff-administrator/components/staff-user-details/staff-user-details.component.ts

```diff
@@ -39,36 +40,55 @@ export class StaffUserDetailsComponent {
   }
 
   public updateUserStatus(): void {
-    if (!this.loading && !this.userDetails.suspended) {
+    if (!this.loading) {
       this.loading = true;
       const staffUser = new StaffUser();
       Object.assign(staffUser, this.userDetails);
-      staffUser.suspended = true;
-      this.staffDataAccessService.updateUser(staffUser).pipe(
-        finalize(() => {
-          this.loading = false;
-          window.scrollTo(0, 0);
-        })
-      )
+      staffUser.suspended = !staffUser.suspended;
+      this.staffDataAccessService.updateUser(staffUser)
         .subscribe(
           () => {
-            this.status = InfoMessageType.SUCCESS;
-            this.title = 'User suspended';
-            this.message = InfoMessage.SUSPEND_USER_SUCCESS;
-            this.userDetails.suspended = staffUser.suspended;
+            this.getUpdatedUser();
           },
           (err) => {
-            if (err.status === 401 || err.status.toString().startsWith('5')) {
-              this.router.navigateByUrl('/service-down');
-            } else {
-              this.status = 'warning';
-              this.message = InfoMessage.SUSPEND_USER_ERROR;
-            }
+            this.setErrorSteps(err);
           }
         );
     }
   }
 
+  // this is order to ensure the status of the user is correct (i.e. not pending or previous status)
+  private getUpdatedUser(): void {
+    this.staffDataAccessService.fetchSingleUserById(this.userDetails.case_worker_id).pipe(
+      map((user) => this.userDetails = StaffUser.from(user)),
```

Comment:
if you're doing side effects; you should be using tap

But I guess it makes more sense to just use the map without this.userDetails = StaffUser.from(user) -- just return StaffUser.from(user). Not very precious about it, but I would add the map operator after the finalize for better readability.

then in the subscribe method; you have access to the user and then you can check 
.subscribe((user) => {
      this.userDetails = user
      if (user.up_idam_status !== StaffUserIDAMStatus.PENDING) {
        this.status = InfoMessageType.SUCCESS;
        // User either restored or suspended if successful
        this.title = user.suspended ? 'User suspended' : 'User restored';
        this.message = user.suspended ? InfoMessage.SUSPEND_USER_SUCCESS : InfoMessage.USER_RESTORED;
      }